src/handle_provide_trusted_info.c

```diff
@@ -0,0 +1,841 @@
+
+#define RCV_FLAG(a) (1 << a)
+
+/*
+trusted_name_descriptor =   tlv(TAG_STRUCTURE_TYPE, u8(TYPE_TRUSTED_NAME))  3 bytes +
+                        &   tlv(TAG_VERSION, u8(0x02))                      3 bytes +
+                        &   tlv(TAG_TRUSTED_NAME_TYPE, 0x06)                3 bytes +
+                        &   tlv(TAG_TRUSTED_NAME_SOURCE, 0x06)              3 bytes +
+                        &   tlv(TAG_TRUSTED_NAME, trusted_name)             2 + 44 bytes +
+                        &   tlv(TAG_CHAIN_ID, chain_id)                     2 + 8 bytes +
+                        &   tlv(TAG_ADDRESS, address)                       2 + 44 bytes +
+                        &   tlv(TAG_SOURCE_CONTRACT, source_contract)*      2 + 44 bytes +
+                        &   tlv(TAG_CHALLENGE, challenge)                   2 + 4 bytes +
+                        &   tlv(TAG_SIGNER_KEY_ID, key_id)                  2 + 2 bytes +
+                        &   tlv(TAG_SIGNER_ALGORITHM, signature_algorithm)  2 + 1 byte +
+                        &   tlv(TAG_SIGNATURE, signature(~,~))              2 + 72 bytes => 247
+
+T  L  V
+01 01 03                                                                TAG_STRUCTURE_TYPE
+02 01 02                                                                TAG_VERSION
+70 01 06                                                                TAG_TRUSTED_NAME_TYPE
+71 01 06                                                                TAG_TRUSTED_NAME_SOURCE
+20 20 276497ba0bb8659172b72edd8c66e18f561764d9c86a610a3a7e0f79c0baf9db  TAG_TRUSTED_NAME
+23 01 65                                                                TAG_CHAIN_ID
+22 20 606501b302e1801892f80a2979f585f8855d0f2034790a2455f744fac503d7b5  TAG_ADDRESS
+73 20 c6fa7af3bedbad3a3d65f36aabc97431b1bbe4c2d2f6e0e47ca60203452f5d61  TAG_SOURCE_CONTRACT
+12 04 deadbeef                                                          TAG_CHALLENGE
+13 01 03                                                                TAG_SIGNER_KEY_ID
+14 01 01                                                                TAG_SIGNER_ALGORITHM
+15 47 30..2e
+
+
+01 01 03
+02 01 02
+70 01 06
+71 01 06
+20 20 276497BA0BB8659172B72EDD8C66E18F561764D9C86A610A3A7E0F79C0BAF9DB
+23 01 65
+22 20 606501B302E1801892F80A2979F585F8855D0F2034790A2455F744FAC503D7B5
+73 20 C6FA7AF3BEDBAD3A3D65F36AABC97431B1BBE4C2D2F6E0E47CA60203452F5D61
+12 04 DEADBEEF
+13 01 00
+14 01 01
+15 47 30..CF
+
+*/
+
+#define TLV_BUFFER_LENGTH 255 /* >= 247, APDU max payload */
+static uint8_t tlv_buffer[TLV_BUFFER_LENGTH] = {0};
+
+typedef enum {
+    STRUCT_TYPE = 0x01,
+    STRUCT_VERSION = 0x02,
+    TRUSTED_NAME_TYPE = 0x70,
+    TRUSTED_NAME_SOURCE = 0x71,
+    TRUSTED_NAME = 0x20,
+    CHAIN_ID = 0x23,
+    ADDRESS = 0x22,
+    SOURCE_CONTRACT = 0x73,
+    CHALLENGE = 0x12,
+    SIGNER_KEY_ID = 0x13,
+    SIGNER_ALGO = 0x14,
+    SIGNATURE = 0x15,
+} e_tlv_tag;
+
+typedef enum { KEY_ID_TEST = 0x00, KEY_ID_PROD = 0x03 } e_key_id;
```

Comment:
`KEY_ID_PROD` is expected to be `0x04`, I tested locally and it works with the current cal
```suggestion
typedef enum { KEY_ID_TEST = 0x00, KEY_ID_PROD = 0x04 } e_key_id;
```